### PR TITLE
Make Borealis GUI more generic to build

### PR DESCRIPTION
**Pull request by thestr4ng3r**

### CODE REVIEW

**Comment by thestr4ng3r** [comment]
cc @H0neyBadger 
With this, there is only one cmake var `CHIAKI_ENABLE_BOREALIS` to determine whether the borealis gui will be built nor not. The Switch/Linux differentiation is then just done depending on which toolchain is built with. This way you can for example also build both Qt gui and borealis in the same build dir.

File: .gitmodules [hunk 1]
```diff
@@ -14,5 +14,5 @@
 	path = android/app/src/main/cpp/oboe
 	url = https://github.com/google/oboe
 [submodule "third-party/borealis"]
```
Comment (H0neyBadger) [inline .gitmodules:16]: the submodule still refer the "third-party" path in its title. 
I do not know if it's intended or not. 
(git submodules can be tricky to move or delete sometimes)

File: CMakeLists.txt [hunk 2]
```diff
@@ -39,33 +39,16 @@ set(CPACK_DEBIAN_PACKAGE_DESCRIPTION ${CPACK_PACKAGE_DESCRIPTION})
 set(CPACK_DEBIAN_PACKAGE_SECTION "games")
 include(CPack)
 
+set(CHIAKI_IS_SWITCH ${NSWITCH})
```
Comment (H0neyBadger) [inline CMakeLists.txt:42]: the NSWICTH flag comes from /opt/devkitpro/switch.cmake
Hope they will not change it in a future release.
maybe duplicate the flag in our local cmake/switch.cmake toolchain might help.

File: switch/src/main.cpp [hunk 3]
```diff
@@ -20,7 +20,7 @@ bool appletMainLoop()
 }
 #endif
 
-#ifndef CHIAKI_SWITCH_ENABLE_LINUX
+#if 0
```
Comment (H0neyBadger) [inline switch/src/main.cpp:23]: the NXLink tool is currently the only way to retrieve logs from the nintendo switch.
Having it enable by default helps "advanced" users to share their logs.

**Review by H0neyBadger — APPROVED** [review]
As always you have done a very good job.

I do not have my switch, ps4 to test the nro but it should not block the release. 
I've added 3 comments. I see no blocking issue. 
I will do a PR later if needed.
thank you a lot ;-)

**Review by H0neyBadger — APPROVED** [review]

**Pull request merged**